Prog/vacant-lots-montgomery_YS.Rmd

```diff
@@ -0,0 +1,1257 @@
+                               " sf"),
+           buildable_area = lotsize_sf * lot_coverage_max,
+           living_area = buildable_area * construction_efficiency,
+           units_per_floor = floor(living_area / average_unit_size),
+           floors = floor(height_ft / 10),
+           units_height= units_per_floor*floors,
+           units_lotsize = floor(lotsize_sf / lotarea_per_unit)) %>% 
+    mutate(subset=as.numeric(floor(lotsize_sf/lot_area_min))) %>% 
+    mutate(units_total= pmin(units_height, units_lotsize, na.rm = TRUE )) %>% 
+    mutate( units_total = ifelse(is.na(units_total)==1|units_total > (density *(lotsize_sf/43560)), floor(density *(lotsize_sf/43560)),units_total))
+
+}
+
+
+#' Get maximum unit summary
+#'
+#' @param dataset dataset with lot size and zoning variables 
+#' @param construction_efficiency Assumption for construction efficiency
+#' @param average_unit_size Assuption for average unit size- should be derived from the American Housing Survey. Given in square feet.
+#' @param dwelling_type_breakdown Options are TRUE or FALSE- indicates whether or not estimates should be broken down by `dwelling_type`
+#'
+#' @return Returns a dataframe with summary statistics on the maximum units that can be build on lots, per the by-right zoning code.
+#'
+#' @examples
+get_unit_summary_g <- function(dataset, 
+                             construction_efficiency, 
+                             average_unit_size,
+                             dwelling_type_breakdown) {
+
+  x <- dataset %>% 
+    get_unit_variables_g(construction_efficiency,
+                       average_unit_size) %>%
+    group_by(parameters) %>% 
+    summarize(lots = n(),
+              units_height = sum(units_height, na.rm = TRUE),
+              units_lotsize = sum(units_lotsize, na.rm = TRUE),
+              units_total = sum(units_total, na.rm = TRUE),
+              units_per_lot = floor(sum(units_total / lots)))
+  
+  y <- dataset %>% 
+    get_unit_variables_g(construction_efficiency,
+                       average_unit_size) %>%
+    group_by(parameters, housing_type) %>% 
+    summarize(lots = n(),
+              units_height = sum(units_height, na.rm = TRUE),
+              units_lotsize = sum(units_lotsize, na.rm = TRUE),
+              units_total = sum(units_total, na.rm = TRUE),
+              units_per_lot = floor(sum(units_total/lots))) %>% 
+    mutate(housing_type = as.character(housing_type))
+  
+  
+  if (dwelling_type_breakdown == TRUE) {
+   
+      bind_rows(y, mutate(x, housing_type = "Sum"))  
+  
+  } else if (dwelling_type_breakdown == FALSE) {
+      x
+    }
+}
+
+
+
+```
+    
+
+Get summary statistics for several different parameters
+
+```{r sum Gaithersburg}
+
+get_unit_summary_g(dataset = zoningdata_gaithersburg1_final,
+                 construction_efficiency = .7,
+                 average_unit_size = 940,
+                 dwelling_type_breakdown = TRUE) %>% 
+  knitr::kable()
```

Comment:
units per lot is coming out to 0 for SF?

Reply:
less than one will be rounded to 0 in the summary table. I added a new variable that's the unrounded unit per lot.